### PR TITLE
cmake: Add seL4Config.cmake include CMakeLists.txt

### PR DESCRIPTION
**Pull request by kent-mcleod**
seL4Config.cmake is responsible for generating a valid
CMAKE_TOOLCHAIN_FILE and setting up platform config options at the start
of the build. The CMAKE_TOOLCHAIN_FILE variable has to be set before the
first cmake `project()` function is processed to take effect.
Previously this file was required to be imported in a CMake script
before the kernel's CMakeLists.txt could be processed. This prevented
the main CMakeLists.txt file from being used without an additional
configuration file:
cmake -G Ninja -C ../configs/ARM_verified.cmake ../

Now it is possible to do:
cmake -G Ninja -DKernelPlatform=imx6 -DKernelARMPlatform=sabre ../

This should make it easier to invoke CMake for building kernel
configurations from other build environments.

Because this file is now imported in the Kernel's CMakeLists.txt
context, there is no longer a requirement to save all the intermediate
settings into the cache and then read them out again.

Signed-off-by: Kent McLeod <kent@kry10.com>

### CODE REVIEW

**Comment by kent-mcleod** [comment]
@ssrg-bamboo `test`

File: configs/seL4Config.cmake [hunk 1]
```diff
@@ -185,7 +156,6 @@ unset(KernelArmCPU CACHE)
 unset(KernelArmArmV CACHE)
 
 # Blacklist platforms without MCS support
```
Comment (axel-h) [inline configs/seL4Config.cmake:158]: maybe this comment should also be removed. Or updated to say that platform must explicitly turn this off now.

Comment (kent-mcleod) [reply configs/seL4Config.cmake:158]: I think your other PR that updates the comment should be sufficient.  I haven't changed the behavior of KernelPlatformSupportsMCS in a way that makes the comment wrong. 
KernelPlatformSupportsMCS still works the same way:
- It's default to ON, but platforms that don't support MCS set it to off.
What's changed is that the result isn't written into the cache which was requiring the entry to be deleted each time the script reruns.

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Review by lsf37 — APPROVED** [review]

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by lsf37** [comment]
@kent-mcleod This is independent of this PR, but the style check wants the `.cmake` files to be not executable. Is there a reason why they were set executable in the first place?

**Comment by lsf37** [comment]
Ok, I found 297d2b63da39ea26b3c437a15247fcc6508d8598, which explains the `+x`. I'm happy to ignore the style check on this one.

**Comment by lsf37** [comment]
Cancelling the hardware test before merge where it will run again anyway (it has passed before the branch update).

**Pull request merged**